src/components/SearchAndFilter/SearchAndFilter.tsx

```diff
@@ -41,22 +41,22 @@ const SearchAndFilter: React.FC<filterProps> = ({allData, columns, filterTypes})
   const createFilterOptions = (filterType: string) => {
     const availableOptions = determineAvailableFilterValues(filterType)
 
-    const options = availableOptions.map((option, i) => {
+    const options = availableOptions.reduce((options:React.ReactNode[], option, i) => {
```

Comment:
This was in order to address that a conditional `map` can still return undefined values if the conditional doesn't pass (although I don't believe I noticed an effect from this for this specific iterator.)

This iterator might do better as a helper function: `removeDuplicateOptions`, but it does contain the logic of the final output to JSX for filter drop downs.